explanations/function_evaluator.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "wealth_grid = processed_model.grids[\"wealth\"]\n",
+    "\n",
+    "plt.plot(wealth_grid, vf_arr)\n",
+    "plt.scatter(wealth_grid, vf_arr)\n",
+    "plt.xlabel(\"Wealth ($x$)\")\n",
+    "plt.ylabel(\"$V_T(x)$\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Interpolation\n",
+    "\n",
+    "What happens now if we want to know the value of $V_T$ at 25 or 75? We need to perform\n",
+    "some kind of interpolation. This is where the function evaluator comes into play, which\n",
+    "returns pre-calculated values if evaluated on a grid point, and linearly interpolated\n",
+    "values otherwise.\n",
+    "\n",
+    "To optimally utilize the structure of the grids when interpolating, the function evaluator\n",
+    "requires information on the state space."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from lcm.state_space import create_state_choice_space\n",
+    "\n",
+    "# the space info object contains information on the grid structure etc.\n",
+    "_, space_info, *_ = create_state_choice_space(\n",
+    "    model=processed_model,\n",
+    "    period=1,\n",
+    "    is_last_period=True,\n",
+    "    jit_filter=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Generating the function evaluator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from functools import partial\n",
+    "\n",
+    "from lcm.function_evaluator import get_function_evaluator\n",
+    "\n",
+    "_value_function = get_function_evaluator(\n",
```

Comment:
Let's try to brainstorm for whether we can find more expressive names here. I find it hard to wrap my head around this cell.

Reply:
How about `scalar_function_evaluator`?